Tests/SKCoreTests/BuildSystemManagerTests.swift

```diff
@@ -412,16 +404,19 @@ final class BuildSystemManagerTests: XCTestCase {
 // MARK: Helper Classes for Testing
 
 /// A simple `MainFilesProvider` that wraps a dictionary, for testing.
-private final class ManualMainFilesProvider: MainFilesProvider {
-  let lock: DispatchQueue = DispatchQueue(label: "\(ManualMainFilesProvider.self)-lock")
-  private var _mainFiles: [DocumentURI: Set<DocumentURI>] = [:]
-  var mainFiles: [DocumentURI: Set<DocumentURI>] {
-    get { lock.sync { _mainFiles } }
-    set { lock.sync { _mainFiles = newValue } }
+private final actor ManualMainFilesProvider: MainFilesProvider {
+  private var mainFilesDictionary: [DocumentURI: Set<DocumentURI>]
+
+  init(mainFilesDictionary: [DocumentURI: Set<DocumentURI>]) {
+    self.mainFilesDictionary = mainFilesDictionary
+  }
```

Comment:
And I think we can make the parameter unnamed here. Since we are constructing a `ManualMainFilesProvider`, it’s clear that we are passing the main files, IMO. What do you think?

Reply:
agree. let me update the code.